### DISCUSSION TITLE
Testing.Platform test runner with solution file

### DISCUSSION THREAD

**Discussion by ladeak** [microsoft/testfx#4105] [Q&A] [answered]
I have a solution with 3 test projects. In the past I could do: `dotnet test MySolution.sln --configuration Release` in a CI.

When I migrate to the TestRunner, what is the best way to execute all tests in the solution? I would like all tests projects to execute but not my other console apps in the solution. Can I avoid listing all test projects?

Something like 
```
dotnet build MySolution.sln --configuration Release
dotnet run ? 
```

**Comment by Evangelink** [comment]
`dotnet test` is still the recommended way to run all your tests. Are you facing some issue with it?

**Reply by ladeak** [reply]
No, but I'd like to try to the new runner. For one of my test projects I measure ~20% improvement, which made me exited enough to try it.

**Reply by Evangelink** [reply, marked as answer]
The new platform does integrate with `dotnet test`, please do check the extra property is turned on (see https://learn.microsoft.com/dotnet/core/testing/unit-testing-platform-integration-dotnet-test#dotnet-test---microsofttestingplatform-mode). If you are using `MSTest.Sdk` then no extra step is required :)

Please keep me posted about the perf you are measuring, it's always good to get more data!

**Reply by ladeak** [reply]
Actually for test projects using xunit. On GitHub they used to before 20 sec, now 14 sec. Similar improvements locally.
before:
https://github.com/ladeak/Http3Tools/actions/runs/11901866566/job/33165707227
after:
https://github.com/ladeak/Http3Tools/actions/runs/11941846098/job/33287693454

I suppose the --no-build helps a few seconds too, but without it with xunit there is a few seconds gain as well.
